test/gateway_composer/utils.js

```diff
@@ -12,23 +12,118 @@
 // See the License for the specific language governing permissions and
 // limitations under the License.
 
-const GatewayComposer = artifacts.require('TestGatewayComposer');
+const BN = require('bn.js');
 
-/**
- * Setup GatewayComposer.
- */
-module.exports.setupGatewayComposer = async (accountProvider) => {
+const MockBrandedTokenPass = artifacts.require('MockBrandedTokenPass');
+const MockBrandedTokenFail = artifacts.require('MockBrandedTokenFail');
+const GatewayComposer = artifacts.require('GatewayComposer');
+const EIP20TokenMock = artifacts.require('EIP20TokenMock');
+const MockGatewayPass = artifacts.require('MockGatewayPass');
+const MockGatewayFail = artifacts.require('MockGatewayFail');
+
+const symbol = 'Test';
+const name = 'Test';
+const decimals = 18;
+const conversionRate = 1;
+const conversionRateDecimals = 0;
+
+module.exports.setupGatewayComposer = async (accountProvider, useBTPass = true) => {
+    const organization = accountProvider.get();
     const owner = accountProvider.get();
-    const valueToken = accountProvider.get();
-    const brandedToken = accountProvider.get();
+    const ownerBalance = new BN(1000);
+
+    const valueToken = await EIP20TokenMock.new(
+        symbol,
+        name,
+        decimals,
+        { from: organization },
+    );
+
+    await valueToken.setBalance(owner, ownerBalance);
+
+    assert.strictEqual(
+        (await valueToken.balanceOf.call(owner)).cmp(ownerBalance),
+        0,
+    );
+
+    let brandedToken;
+    if (useBTPass === true) {
+        brandedToken = await this.setupBrandedTokenPass(valueToken, organization);
+    } else {
+        brandedToken = await this.setupBrandedTokenFail(valueToken, organization);
+    }
 
     const gatewayComposer = await GatewayComposer.new(
         owner,
+        valueToken.address,
+        brandedToken.address,
+    );
+
+    return {
         valueToken,
         brandedToken,
+        gatewayComposer,
+        owner,
+        organization,
+    };
+};
+
+module.exports.setupBrandedTokenPass = async (valueToken, organization) => {
+    const brandedToken = await MockBrandedTokenPass.new(
+        valueToken.address,
+        symbol,
+        name,
+        decimals,
+        conversionRate,
+        conversionRateDecimals,
+        organization,
+    );
+
+    return brandedToken;
+};
+
+module.exports.setupBrandedTokenFail = async (valueToken, organization) => {
+    const brandedToken = await MockBrandedTokenFail.new(
+        valueToken.address,
+        symbol,
+        name,
+        decimals,
+        conversionRate,
+        conversionRateDecimals,
+        organization,
+    );
+
+    return brandedToken;
+};
+
+module.exports.setupGatewayComposerRequestStake = async (valueToken, gatewayComposer, owner) => {
```

Comment:
Suggest renaming this function, as the name implies to me that this function will request a stake and not setup GC to be able to successfully execute an impending call to request a stake.